runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1443,6 +1477,31 @@ gcParseCommandLineAndInitializeWithValues(J9JavaVM *vm, IDATA *memoryParameters)
 	}
 	memoryParameters[opt_Xmx] = index;
 
+	result = option_set_to_opt_double(vm, OPT_XXMAXRAMPERCENT, &index, EXACT_MEMORY_MATCH, &maxRAMPercent);
+	if (OPTION_OK != result) {
+		goto _error;
+	} else {
+		if (-1 != index) {
+			if (maxRAMPercent < 0) {
+				result = OPTION_UNDERFLOW;
+				goto _error;
+			} else if (maxRAMPercent > 100) {
+				result = OPTION_OVERFLOW;
+				goto _error;
+			} else {
+				/* Use -XX:MaxRAMPercentage only if -Xmx is not specified */
+				if (memoryParameters[opt_Xmx] == -1) {
+					OMRPORT_ACCESS_FROM_J9PORT(vm->portLibrary);
+					uint64_t usableMemory = omrsysinfo_get_addressable_physical_memory();
+					extensions->memoryMax = (usableMemory / 100) * maxRAMPercent;
+
+					/* Hack table to appear that -Xmx was specified */
+					memoryParameters[opt_Xmx] = index;
```

Comment:
I don't like this solution. I believe the option should be parsed and stored in gc extensions. MemoryMax and initialMemorySize should be adjusted in mminit. I would try to avoid any decisions in parse module